pkg/api/convertertoapi.go

```diff
@@ -217,7 +223,12 @@ func convertV20170131Properties(v20170131 *v20170131.Properties, api *Properties
 	api.AgentPoolProfiles = []AgentPoolProfile{}
 	for _, p := range v20170131.AgentPoolProfiles {
 		apiProfile := &AgentPoolProfile{}
-		convertV20170131AgentPoolProfile(&p, apiProfile)
+		// api.OrchestratorProfile already be filled in correctly
+		if api.OrchestratorProfile.IsKubernetes() {
```

Comment:
I'm guessing this was merged but If you had mentioned the version being the reason we only allow VMAS for k8s it would of answered @weinong 's question.